public/app/features/transformers/editors/ConvertFieldTypeTransformerEditor.tsx

```diff
@@ -129,6 +129,14 @@ export const ConvertFieldTypeTransformerEditor = ({
     <>
       {options.conversions.map((c: ConvertFieldTypeOptions, idx: number) => {
         const targetField = findField(input?.[0], c.targetField);
+
+        // Show "Join with" input when:
+        // - A join value exists (maintains backward compatibility)
+        // - Target field type is 'other' (Grafana 10) or 'string' (Grafana 11)
+        // This ensures consistent UI across versions where arrays may be classified differently.
+        const shouldRenderJoinWith =
+          c.joinWith?.length || targetField?.type === FieldType.other || targetField?.type === FieldType.string;
```

Comment:
I suggest a construction like this to check among a list of options in an enum, mainly because if you want to add further values in that enum in the future or add more nuance to this condition in some other way, you're not forced to tack more conditional operators at the end for further enums that apply here.

```suggestion
          c.joinWith?.length || [FieldType.other, FieldType.string].includes(targetField?.type);
```

Reply:
Updated!